app/controllers/v3/auth/federation_controller.rb

```diff
@@ -21,6 +25,16 @@ def auth_response(type, *filters)
         headers[x_subject_token_header_key] = Utils::Tokenator.to_token(credentials.to_hash)
         respond_with token_response
       end
+
+      def audit_unscoped_tokens
+        Rails.configuration.x.audit.info "Unscoped token #{digest_response_token} (digest) " \
+                                         "from IP #{request.remote_ip} " \
+                                         "for credentials #{credentials}"
+      end
+
+      def digest_response_token
```

Comment:
Method does pretty much the same thing as method `digest_response_token` in `V3::Auth::TokensController`. Unite them. Preferably in some concern.